torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
 
         return unspec_var
 
+    def wrap_symfloat(self, value):
+        # SymFloat wrapping is special.  We first wrap it in the same way we
+        # do an unspecialized primitive, and then we item() it into a
+        # SymFloat.  Removal of the item() call is left to a later FX pass,
+        # mostly because that pass is more easily done after decomposition
+        # (Dynamo doesn't do decomposition right now).
+
+        if self.name in self.tx.output.unspec_variable_map:
+            return self.tx.output.unspec_variable_map[self.name]
+
+        # NB: we specialize on nan input, because our guard modeling in
+        # ShapeEnv cannot deal with nan
```

Comment:
This should probably be handled in the `ShapeEnv` similar to how we handle 0/1 specialisation. Otherwise if we first trace a float value dynamically without restrictions and then we pass a `NaN`, then we end up with a `NaN` in a variable in the program.

Reply:
I do agree this is an omission right now.